mypy/subtypes.py

```diff
@@ -125,6 +125,10 @@ def visit_instance(self, left: Instance) -> bool:
                                                  self.right,
                                                  self.check_type_parameter):
                 return True
+            if left.type.typeddict_type is not None and is_subtype(left.type.typeddict_type,
```

Comment:
Hmm this likely isn't right. We must assume that a typed dict type is always represented as `TypedDictType` here. The way to ensure this is probably by updating `TypeAnalyser.visit_unbound_type` to perform a check on `typeddict_type` and return one as needed. It already does this for `tuple_type` (towards the end of the method), and this should be similar.